### PR TITLE
Add email information when sharing

### PR DESCRIPTION
**Pull request by GVodyanov**
Fix https://github.com/nextcloud/calendar/issues/4406

TODO:
- [x] Add email in NcSelect
- [x] Add UI for email in `ShareItem.vue`
- [x] Add principal fetching for getting email in `ShareItem.vue`

### CODE REVIEW

File: src/components/AppNavigation/EditCalendarModal/ShareItem.vue [hunk 1]
```diff
@@ -133,6 +140,19 @@ export default {
 				this.updatingSharee = false
 			}
 		},
+
+		async updateShareeEmail() {
+			if (this.sharee.isGroup || this.sharee.isCircle) return
```
Comment (ChristophWurst) [inline src/components/AppNavigation/EditCalendarModal/ShareItem.vue:145]: coding style: use a block for if, not one-liners please

File: src/components/AppNavigation/EditCalendarModal/ShareItem.vue [hunk 2]
```diff
@@ -133,6 +140,19 @@ export default {
...
+
+			const shareeUrl = this.sharee.uri.replace('principal:', '/remote.php/dav/') + '/'
+
+			await this.$store.dispatch('fetchPrincipalByUrl', {url: shareeUrl})
+
+			const principal = this.$store.getters.getPrincipalByUrl(shareeUrl)
+			debugger
```
Comment (ChristophWurst) [inline src/components/AppNavigation/EditCalendarModal/ShareItem.vue:152]: :eyes:

File: src/components/AppNavigation/EditCalendarModal/ShareItem.vue [hunk 3]
```diff
@@ -153,6 +173,12 @@ export default {
 
 	&__label {
 		flex: 1 auto;
+		flex-direction: column;
+
+		p {
+			color: var(--color-text-lighter);
+			line-height: 1;
```
Comment (ChristophWurst) [inline src/components/AppNavigation/EditCalendarModal/ShareItem.vue:180]: that seems odd. does it miss a unit?

Comment (GVodyanov) [reply src/components/AppNavigation/EditCalendarModal/ShareItem.vue:180]: No this is fine, it's based on a proportion to the font size, and the styling is copied from a similar component

**Pull request merged**